### ISSUE TITLE
[JIT] Huge delay (1274s vs 0.031s) when running scripted model

### ISSUE THREAD

**Issue by IlyaOvodov** [pytorch/pytorch#36040]
## 🐛 Bug

When a model that worked properely on a pure python is scripted, it works with a huge delay (1274s vs 0.031s before scripting), while the result at last is the same.

Below are timings for processing 2 copies of the same image. Note that sometimes it is the 1st image that is processed very slowly, while sometimes it is the 2nd one. I've tested 2 models: the simple, wherer delay is about 10s vs 0.008s and more complex where delay is about 1000s vs 0.056s, tests were done on 2 environments: latest nightly pytorch 1.5.0a0+2dc2933 on 1080Ti and on pytorch 1.4.0 on NVIDIA 1050.

Results:

1.5.0a0+2dc2933 on 1080Ti
cfg/yolov3-tiny.cfg weights/yolov3-tiny.pt (more simple model)
Model Summary: 37 layers, 8.85237e+06 parameters, 8.85237e+06 gradients
before scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (0.748s)
image 2/2 data\samples\TEST_AO_2_2_4300002.jpeg: Done. (0.008s) _(...to compare with...)_
after scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (0.170s)
image 2/2 data\samples\TEST_AO_2_2_4300002.jpeg: Done. (11.391s) **(not good...)**

cfg/yolov3.cfg weights/yolov3.pt (more complex model)
Model Summary: 222 layers, 6.19491e+07 parameters, 6.19491e+07 gradients
before scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (0.768s)
image 2/2 data\samples\TEST_AO_2_2_4300001copy.jpeg: Done. (0.031s) _(...to compare with...)_
after scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (1.392s)
image 2/2 data\samples\TEST_AO_2_2_4300001copy.jpeg: Done. (579.728s) **(... terrible)**

1.4.0 on 1050:
cfg/yolov3-tiny.cfg weights/yolov3-tiny.pt (more simple)
Model Summary: 37 layers, 8.85237e+06 parameters, 8.85237e+06 gradients
before scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (0.862s) _(...to compare with...)_
image 2/2 data\samples\TEST_AO_2_2_4300001copy.jpeg: Done. (0.014s)
after scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (5.997s)  **(not good...)**
image 2/2 data\samples\TEST_AO_2_2_4300001copy.jpeg: Done. (0.012s)

cfg/yolov3.cfg weights/yolov3.pt (more complex model)
Model Summary: 222 layers, 6.19491e+07 parameters, 6.19491e+07 gradients
before scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (1.009s) _(...to compare with...)_
image 2/2 data\samples\TEST_AO_2_2_4300001copy.jpeg: Done. (0.056s)
after scripting:
image 1/2 data\samples\TEST_AO_2_2_4300001.jpeg: Done. (1274.444s)  **(... terrible)**
image 2/2 data\samples\TEST_AO_2_2_4300001copy.jpeg: Done. (0.053s)

## To Reproduce
Sorry that I didn't produce more localized example. 
1. git clone https://github.com/IlyaOvodov/yolov3.git
2. git checkout performance_problem_test
3. Download weights yolov3-tiny.pt (for simple model) and yolov3.pt (for complex one) from https://drive.google.com/drive/folders/1LezFG5g3BCW6iYaV89B2i64cqEUZD7e0
and place them into weights directory
4. python test_timing.py
This run a simple model test (with has a 5-10s delay)
For a complex model test uncomment line 26 in test_timing.py: `cfg, weights = ('cfg/yolov3.cfg', 'weights/yolov3.pt')`

## Environment
1.
PyTorch version: 1.5.0a0+2dc2933
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: Microsoft Windows 10 Pro
GCC version: Could not collect
CMake version: version 3.12.2

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 441.87
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip3] numpy==1.17.2
[pip3] pytorch-ignite==0.2.0
[pip3] pytorch-toolbelt==0.2.1
[pip3] torch==1.5.0a0+2dc2933
[pip3] torchvision==0.4.1a0+d94043a
[conda] blas                      1.0                         mkl
[conda] cuda92                    1.0                           0    pytorch
[conda] mkl                       2019.0                   pypi_0    pypi
[conda] mkl-include               2019.0                   pypi_0    pypi
[conda] mkl_fft                   1.0.6            py36hdbbee80_0
[conda] mkl_random                1.0.1            py36h9258bd6_0
[conda] numpy                     1.17.2                   pypi_0    pypi
[conda] pytorch-ignite            0.2.0                     dev_0    <develop>
[conda] pytorch-toolbelt          0.2.1                     dev_0    <develop>
[conda] torch                     1.5.0a0+2dc2933           dev_0    <develop>
[conda] torchvision               0.4.1a0+d94043a           dev_0    <develop>

2.
Collecting environment information...
PyTorch version: 1.4.0
Is debug build: No
CUDA used to build PyTorch: 10.1

OS: Microsoft Windows 7 Џа®дҐббЁ®­ «м­ п
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: GeForce GTX 1050
Nvidia driver version: 441.22
cuDNN version: C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.0\bin\cudnn64_7.dll

Versions of relevant libraries:
[pip3] efficientnet-pytorch==0.6.3
[pip3] numpy==1.18.0
[pip3] torch==1.4.0
[pip3] torchvision==0.5.0
[conda] efficientnet-pytorch      0.6.3                     <pip>
[conda] mkl                       2019.0                    <pip>
[conda] mkl-include               2019.0                    <pip>
[conda] numpy                     1.18.0                    <pip>
[conda] torch                     1.4.0                     <pip>
[conda] torchvision               0.5.0                     <pip>

cc @ezyang @gchanan @zou3519 @suo

**Comment by suo** [comment]
Thanks for the report, and the repro! I'm taking a look at this, we've received similar reports on large models.

**Comment by IlyaOvodov** [comment]
See useful discussion [here](https://github.com/pytorch/pytorch/pull/36345#issuecomment-612789481) , about miraculous cheat code
```
with torch.jit.optimized_execution(false):
    my_model(inputs)
```
etc.

**Comment by suo** [comment]
A similar RAII-style resource guard is available in C++:
```
torch::jit::GraphOptimizerEnabledGuard g(false);
```
Good point about documentation, I'll find some time to add it. (alternatively, I would accept a PR that did so 😛 )

**Comment by IlyaOvodov** [comment]
@suo, It seems to be very questionable optimization that takes 40 minutes (OK, 10 minutes after your enhancements) on application launce to get 3-4% performance boosting running. Why not to set in off by default? 

I've made some tests (sorry for do it without your boosting, but I don't expect it changes opnimized/not optimazed ratio). Columns: Pure - pure python inference (time), NoOpt  - inference on scripted model in python with torch.jit.optimized_execution(false), Opt - inference on scripted model in python as usual (without torch.jit.optimized_execution(false)), Opt/NoOpt - performance boosting.

Light version
Pure | NoOpt | Opt | Opt/NoOpt
0.018 | 0.289 | 21.038 | 72.80
0.016 | 0.014 | 0.012 | 0.86
0.281 | 0.28 | 0.279 | 1.00
0.041 | 0.042 | 0.036 | 0.86
0.072 | 0.047 | 0.044 | 0.94
0.106 | 0.082 | 0.094 | 1.15
0.078 | 0.068 | 0.068 | 1.00
  |   | mean | 0.97
More complex version |   |   |  
Pure | NoOpt | Opt | Opt/NoOpt
0.061 | 3.489 | 2512 | 719.98
0.055 | 0.055 | 0.056 | 1.02
0.333 | 0.33 | 0.31 | 0.94
0.114 | 0.116 | 0.112 | 0.97
0.091 | 0.096 | 0.089 | 0.93
0.113 | 0.114 | 0.107 | 0.94
0.103 | 0.099 | 0.097 | 0.98
  |   | mean | 0.96

**Comment by suo** [comment]
@IlyaOvodov, we are totally in agreement that the current compilation time is too slow. I am working on further improvements currently.

As for why not turn it off by default: your model is tripping some bad behavior in our compiler, but in most cases compile time is not as long. In those cases (where the warm up time is ~less than a minute). We are disabling this bad behavior in the next release, but optimizations as a whole will stay on.

Thanks for bringing this to our attention. I'm going to increase the priority of this issue so we keep track of this and keep it updated.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
